### DISCUSSION TITLE
sha in vulnerabilityreport

### DISCUSSION THREAD

**Discussion by kfox1111** [aquasecurity/starboard#664] [Development]
It would be nice if the sha1 of the container image that was scanned could be stored in the vulnerability report. Possibly as a label on the vulnerability report. Could be used to find instances of an image that was already scanned, or pods that don't end up using the same sha, but the same tag.

**Comment by danielpacak** [comment]
👋 @kfox1111 that's a very good point and I think we already have #298 to track similar requirements.

IMO just labeling instances of VulnerabilityReports with image digest is not enough due to naming convention and ownership that we put in place. In the following listing you can see that we have only one VulnerabilityReport (`replicaset-nginx-db749865c-nginx`) even though there are three replica Pods (`nginx-db749865c-lfcp5`, `nginx-db749865c-tn5k7`, and `nginx-db749865c-vjlr9`).

```console
$ kubectl tree deploy nginx
NAMESPACE  NAME                                                       READY  REASON  AGE
default    Deployment/nginx                                           -              2m22s
default    ├─ReplicaSet/nginx-6d4cf56db6                              -              2m22s
default    │ ├─ConfigAuditReport/replicaset-nginx-6d4cf56db6          -              2m17s
default    │ └─VulnerabilityReport/replicaset-nginx-6d4cf56db6-nginx  -              2m2s
default    └─ReplicaSet/nginx-db749865c                               -              75s
default      ├─ConfigAuditReport/replicaset-nginx-db749865c           -              73s
default      ├─VulnerabilityReport/replicaset-nginx-db749865c-nginx   -              65s
default      ├─Pod/nginx-db749865c-lfcp5                              True           75s
default      ├─Pod/nginx-db749865c-tn5k7                              True           12s
default      └─Pod/nginx-db749865c-vjlr9                              True           12s
```

FYI Here's a POC that we did some time back to see how to detect "mutant tags" https://github.com/danielpacak/k8s-mutant-detector. We can possibly integrate it with Starboard.
